apps/explorer/src/app/components/txs/lib/deterministic-ids.ts

```diff
@@ -0,0 +1,39 @@
+import { sha3_256 } from 'js-sha3';
+
+/**
+ * Encodes a string as bytes
+ * @param hex
+ * @returns number[]
+ */
+export function hexToString(hex: string) {
+  if (!hex.match(/^[0-9a-fA-F]+$/)) {
+    throw new Error('is not a hex string.');
+  }
+  if (hex.length % 2 !== 0) {
+    hex = '0' + hex;
```

Comment:
comment (blocking?): i think it's generally considered a bad practice to silently modify a function's input parameter. maybe you could create a new variable, something like: 
```
const normalizedHex = hex.length % 2 !== 0 ? `0${hex}` : hex;
```

Reply:
Yes - I will update this too